buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -109,18 +108,26 @@ public void apply(Project project) {
 
         // 3. Apply interface injections after the ATs
         // this jar is only used for the patches in the repo
-        var applyInterfaceInjection = configureInterfaceInjection(
-                project, configurations,
-                applyAt.flatMap(ApplyAccessTransformer::getOutputJar),
-                neoDevBuildDir,
-                List.of(project.getRootProject().file("src/main/resources/META-INF/interfaces.json"))
-        );
+        var applyInterfaceInjection = project.getTasks().register("applyInterfaceInjection", TransformSources.class, task -> {
+            task.getInputJar().set(applyAt.flatMap(TransformSources::getOutputJar));
+            task.getInterfaceInjectionData().from(project.getRootProject().file("src/main/resources/META-INF/interfaces.json"));
+            task.getOutputJar().set(neoDevBuildDir.map(dir -> dir.file("artifacts/interface-injected-sources.jar")));
+            task.getInterfaceStubsJar().set(neoDevBuildDir.map(dir -> dir.file("artifacts/interface-injection-stubs.jar")));
```

Comment:
Add comment saying that they are unused but can't be disabled in JST or something.